Sources/IdentityState.swift

```diff
@@ -142,4 +145,19 @@ class IdentityState {
         identityProperties.saveToPersistence()
         createXDMSharedState(identityProperties.toXdmData(), event)
     }
+
+    /// Check if the Identity direct extension is registered by checking the EventHub's shared state list of registered extensions.
+    /// - Parameters:
+    ///   - event: the tiggering event
+    ///   - getSharedState: function to get shared states from the EventHub
+    /// - Returns: true if the Identity direct extension is registered with the EventHub
+    private func isIdentityDirectRegistered(event: Event, getSharedState: @escaping (_ name: String, _ event: Event?, _ barrier: Bool) -> SharedStateResult?) -> Bool {
+        if let registeredExtensionsWithHub = getSharedState(IdentityConstants.SharedState.Hub.SHARED_OWNER_NAME, event, false)?.value,
```

Comment:
Should we rather pass event nil to fetch latest at the time we do this check as I was discussing offline with Nick?

Reply:
Yes, I changed this to get the latest shared state.